src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -227,10 +230,7 @@ public void ConcurrentSendReceive()
                 Task.WaitAll(writes);
                 Task.WaitAll(reads);
 
-                for (int i = 0; i < sendData.Length; i++)
-                {
-                    Assert.True(sendData[i] == receiveData[i], $"Different at {i}");
-                }
+                Assert.Equal(sendData.OrderBy(i => i), receiveData.OrderBy(i => i));
```

Comment:
> SendAsync will send all of the data, unless there's a timeout

So it returns a ```Task<int>``` because of the timeout case?

> ReceiveAsync might return partial results

I'm taking the easy way out and updating the test to use more tasks, but each sending/receiving a single byte.